### ISSUE TITLE
Potential issue with lockBehavior: API v4 and v5 

### ISSUE THREAD

**Issue by arvindsv** [gocd/gocd#3958]
This is just for the record. This was discussed in https://github.com/gocd/gocd/pull/3943#pullrequestreview-71141523 and https://github.com/gocd/gocd/pull/3943#issuecomment-339751913 and decided that it's a "WONTFIX".

Once https://github.com/gocd/gocd/pull/3943 is merged,  PR v4 continues to work as is, that is it return booleans for `isLocked` and v5 starts allowing one of the three values to be specified for the `lockBehavior` attribute.

Consider this scenario:
1. Person 1: Update the pipeline config from quick-edit/old-UI/apiv5 to set it to `unlockWhenFinished`
2. Person 2: Queries for the config using apiv4 (returns the boolean value, false in this case), edits a totally unrelated field say material and updates the config.

Result: This would unintentionally revert the changes made by Person 1.

Resolution: This will be marked as WONTFIX, because it is doing the right thing, but can be confusing for someone who comes across this. The recommendation is to not use an older version of the API.

**Comment by arvindsv** [comment]
Marking as WONTFIX, as mentioned above.
